api/krusty/fnplugin_test.go

```diff
@@ -504,313 +504,94 @@ func skipIfNoDocker(t *testing.T) {
 }
 
 func TestFnContainerGenerator(t *testing.T) {
-	t.Skip("wait for #3881")
 	skipIfNoDocker(t)
-
-	// Function plugins should not need the env setup done by MakeEnhancedHarness
 	th := kusttest_test.MakeHarness(t)
-
-	th.WriteK(".", `
-resources:
-- short_secret.yaml
+	o := th.MakeOptionsPluginsEnabled()
+	fSys := filesys.MakeFsOnDisk()
+	b := MakeKustomizer(&o)
+	tmpDir, err := filesys.NewTmpConfirmedDir()
+	assert.NoError(t, err)
+	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(`
 generators:
-- gener.yaml
-`)
-	// Create generator config
-	th.WriteF("gener.yaml", `
-apiVersion: examples.config.kubernetes.io/v1beta1
-kind: CockroachDB
+- project-service-set.yaml
+`)))
+	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "project-service-set.yaml"), []byte(`
+apiVersion: blueprints.cloud.google.com/v1alpha1
+kind: ProjectServiceSet
 metadata:
   name: demo
   annotations:
     config.kubernetes.io/function: |
       container:
-        image: gcr.io/kustomize-functions/example-cockroachdb:v0.1.0
+        image: gcr.io/kpt-fn/enable-gcp-services:v0.1.0
 spec:
-  replicas: 3
-`)
-	// Create some additional resource just to make sure everything is added
```

Comment:
Please preserve this aspect of the test, i.e. have an existing resource that should continue to appear in the result

Reply:
That sounds reasonable. 
I added another resource to assure this aspect when applying a generator.
https://github.com/kubernetes-sigs/kustomize/pull/5111/commits/c9f500cc0b4b038541629bab9e578acdc3921c08